### ISSUE TITLE
orgstat is broken by base 4.14

### ISSUE THREAD

**Issue by sternenseemann** [volhovM/orgstat#15]
On current `haskell-updates`:

```
> env NIXPKGS_ALLOW_BROKEN=1 nix-build -E 'with (import ./. {}); haskell.lib.doJailbreak haskellPackages.orgstat'
these derivations will be built:
  /nix/store/gk8fb9ksd0z464zbcvb164aabrpq94nc-orgstat-0.1.9.drv
building '/nix/store/gk8fb9ksd0z464zbcvb164aabrpq94nc-orgstat-0.1.9.drv'...
setupCompilerEnvironmentPhase
Build with /nix/store/qabwi7rkqwppxbp575vq7khy6v2vqv81-ghc-8.10.4.
unpacking sources
unpacking source archive /nix/store/nzas24ghpz2ncikvk0zr6krshvygihw8-orgstat-0.1.9.tar.gz
source root is orgstat-0.1.9
setting SOURCE_DATE_EPOCH to timestamp 1609438003 of file orgstat-0.1.9/CHANGES.md
patching sources
Run jailbreak-cabal to lift version restrictions on build inputs.
compileBuildDriverPhase
setupCompileFlags: -package-db=/build/setup-package.conf.d -j4 +RTS -A64M -RTS -threaded -rtsopts
[1 of 1] Compiling Main             ( Setup.hs, /build/Main.o )
Linking Setup ...
configuring
configureFlags: --verbose --prefix=/nix/store/v61wf8h5cgan84y3qryvrs2qiqnq5vim-orgstat-0.1.9 --libdir=$prefix/lib/$compiler --libsubdir=$abi/$libname --docdir=/nix/store/c734gdxwq1s83w1hlq6dwzq2rppcmp14-orgstat-0.1.9-doc/share/doc/orgstat-0.1.9 --with-gcc=gcc --package-db=/build/package.conf.d --ghc-options=-j4 +RTS -A64M -RTS --disable-split-objs --enable-library-profiling --profiling-detail=exported-functions --disable-profiling --enable-shared --disable-coverage --enable-static --disable-executable-dynamic --enable-tests --disable-benchmarks --enable-library-vanilla --disable-library-for-ghci --ghc-option=-split-sections --extra-lib-dirs=/nix/store/4jllf2xc8cpk25cqmnfkhf3xhfdmgrjk-ncurses-6.2/lib --extra-lib-dirs=/nix/store/j924jn8sw7kggc773wllw3kmimmr8z4x-libffi-3.3/lib --extra-lib-dirs=/nix/store/hm6wxjrn2yac2l41n1zkxbdkiw7njr6g-gmp-6.2.1/lib
Using Parsec parser
Configuring orgstat-0.1.9...
Dependency aeson -any: using aeson-1.5.6.0
Dependency ansi-terminal -any: using ansi-terminal-0.11
Dependency attoparsec -any: using attoparsec-0.13.2.5
Dependency base -any: using base-4.14.1.0
Dependency boxes -any: using boxes-0.1.5
Dependency bytestring -any: using bytestring-0.10.12.0
Dependency colour -any: using colour-2.3.5
Dependency containers -any: using containers-0.6.2.1
Dependency data-default -any: using data-default-0.7.1.1
Dependency diagrams-lib -any: using diagrams-lib-1.4.3
Dependency diagrams-svg -any: using diagrams-svg-1.4.3
Dependency directory -any: using directory-1.3.6.0
Dependency exceptions -any: using exceptions-0.10.4
Dependency filepath -any: using filepath-1.4.2.1
Dependency fmt -any: using fmt-0.6.1.2
Dependency formatting -any: using formatting-6.3.7
Dependency hashable -any: using hashable-1.3.0.0
Dependency lens -any: using lens-4.19.2
Dependency mtl -any: using mtl-2.2.2
Dependency optparse-simple -any: using optparse-simple-0.1.1.3
Dependency orgmode-parse -any: using orgmode-parse-0.2.3
Dependency process -any: using process-1.6.9.0
Dependency text -any: using text-1.2.4.1
Dependency time -any: using time-1.9.3
Dependency turtle -any: using turtle-1.5.21
Dependency universum -any: using universum-1.7.2
Dependency yaml -any: using yaml-0.11.5.0
Dependency base -any: using base-4.14.1.0
Dependency bytestring -any: using bytestring-0.10.12.0
Dependency directory -any: using directory-1.3.6.0
Dependency exceptions -any: using exceptions-0.10.4
Dependency filepath -any: using filepath-1.4.2.1
Dependency formatting -any: using formatting-6.3.7
Dependency optparse-simple -any: using optparse-simple-0.1.1.3
Dependency orgstat -any: using orgstat-0.1.9
Dependency universum -any: using universum-1.7.2
Dependency attoparsec -any: using attoparsec-0.13.2.5
Dependency base -any: using base-4.14.1.0
Dependency bytestring -any: using bytestring-0.10.12.0
Dependency directory -any: using directory-1.3.6.0
Dependency exceptions -any: using exceptions-0.10.4
Dependency filepath -any: using filepath-1.4.2.1
Dependency formatting -any: using formatting-6.3.7
Dependency optparse-simple -any: using optparse-simple-0.1.1.3
Dependency orgmode-parse -any: using orgmode-parse-0.2.3
Dependency orgstat -any: using orgstat-0.1.9
Dependency text -any: using text-1.2.4.1
Dependency time -any: using time-1.9.3
Dependency turtle -any: using turtle-1.5.21
Dependency universum -any: using universum-1.7.2
Dependency unordered-containers -any: using unordered-containers-0.2.13.0
Dependency HUnit -any: using HUnit-1.6.2.0
Dependency QuickCheck -any: using QuickCheck-2.14.2
Dependency base -any: using base-4.14.1.0
Dependency colour -any: using colour-2.3.5
Dependency hspec -any: using hspec-2.7.8
Dependency lens -any: using lens-4.19.2
Dependency orgstat -any: using orgstat-0.1.9
Dependency quickcheck-text -any: using quickcheck-text-0.1.2.1
Dependency text -any: using text-1.2.4.1
Dependency time -any: using time-1.9.3
Dependency transformers -any: using transformers-0.5.6.2
Dependency universum -any: using universum-1.7.2
Source component graph:
    component lib
    component exe:orgstat dependency lib
    component exe:orgstatarch dependency lib
    component test:orgstat-test dependency lib
Configured component graph:
    component orgstat-0.1.9-Irl6FpkNJ5D124nqwx7n2s
        include aeson-1.5.6.0-EJQ3uwDcV54EcRrluu2VE
        include ansi-terminal-0.11-9FsdYO0Hps83uFNLVimBZk
        include attoparsec-0.13.2.5-3HlU1muxd0r83R7NHnGRSw
        include base-4.14.1.0
        include boxes-0.1.5-6yGh7Ph1Utu8wxVXCPSyeh
        include bytestring-0.10.12.0
        include colour-2.3.5-C29WD8Qg69AL3qeQgayHy7
        include containers-0.6.2.1
        include data-default-0.7.1.1-CYtyoeRTyO02nIfJy16t2P
        include diagrams-lib-1.4.3-4T3pMpxqmsxGUSWBTZP2Yh
        include diagrams-svg-1.4.3-FUc4JukGvmFI4bl5xnYPyB
        include directory-1.3.6.0
        include exceptions-0.10.4
        include filepath-1.4.2.1
        include fmt-0.6.1.2-D5ieS3bbHQnG2n449eoYuO
        include formatting-6.3.7-8emLtPUrsOY3pXsJkNgaCF
        include hashable-1.3.0.0-KzryYcDwvBoH5VSVaB6Zwd
        include lens-4.19.2-7Erpllp4x6QJde6QNgn5al
        include mtl-2.2.2
        include optparse-simple-0.1.1.3-GCsoEruwqgiElMemPrtAbu
        include orgmode-parse-0.2.3-GD8YuLIKYQFIZjgiqbOh4W
        include process-1.6.9.0
        include text-1.2.4.1
        include time-1.9.3
        include turtle-1.5.21-IR6Ez24zwnKHcviOCCfKnL
        include universum-1.7.2-4C67qAn8hpSBO1f4hPkb4C
        include yaml-0.11.5.0-GXKwO3znmdcIxQZ2p4lFUb
    component orgstat-0.1.9-9vAWSyx0oj1GxDiGvBvCm0-orgstat
        include base-4.14.1.0
        include bytestring-0.10.12.0
        include directory-1.3.6.0
        include exceptions-0.10.4
        include filepath-1.4.2.1
        include formatting-6.3.7-8emLtPUrsOY3pXsJkNgaCF
        include optparse-simple-0.1.1.3-GCsoEruwqgiElMemPrtAbu
        include orgstat-0.1.9-Irl6FpkNJ5D124nqwx7n2s
        include universum-1.7.2-4C67qAn8hpSBO1f4hPkb4C
    component orgstat-0.1.9-BZzeApBR7Mk86xSdbWZLOo-orgstatarch
        include attoparsec-0.13.2.5-3HlU1muxd0r83R7NHnGRSw
        include base-4.14.1.0
        include bytestring-0.10.12.0
        include directory-1.3.6.0
        include exceptions-0.10.4
        include filepath-1.4.2.1
        include formatting-6.3.7-8emLtPUrsOY3pXsJkNgaCF
        include optparse-simple-0.1.1.3-GCsoEruwqgiElMemPrtAbu
        include orgmode-parse-0.2.3-GD8YuLIKYQFIZjgiqbOh4W
        include orgstat-0.1.9-Irl6FpkNJ5D124nqwx7n2s
        include text-1.2.4.1
        include time-1.9.3
        include turtle-1.5.21-IR6Ez24zwnKHcviOCCfKnL
        include universum-1.7.2-4C67qAn8hpSBO1f4hPkb4C
        include unordered-containers-0.2.13.0-LfX1QsW45a49ihvoGPwDYs
    component orgstat-0.1.9-DitnakZGhXaFkCDXafAtcx-orgstat-test
        include HUnit-1.6.2.0-H4jVFZwCpZD3gLDGn5X1iO
        include QuickCheck-2.14.2-emyW1cBTLx2Wxf2sOiMGt
        include base-4.14.1.0
        include colour-2.3.5-C29WD8Qg69AL3qeQgayHy7
        include hspec-2.7.8-Aj0ANgyNRIsCgvQsvGCAl8
        include lens-4.19.2-7Erpllp4x6QJde6QNgn5al
        include orgstat-0.1.9-Irl6FpkNJ5D124nqwx7n2s
        include quickcheck-text-0.1.2.1-9ONdHqB9s1mEZeV5Jvlbp8
        include text-1.2.4.1
        include time-1.9.3
        include transformers-0.5.6.2
        include universum-1.7.2-4C67qAn8hpSBO1f4hPkb4C
Linked component graph:
    unit orgstat-0.1.9-Irl6FpkNJ5D124nqwx7n2s
        include aeson-1.5.6.0-EJQ3uwDcV54EcRrluu2VE
        include ansi-terminal-0.11-9FsdYO0Hps83uFNLVimBZk
        include attoparsec-0.13.2.5-3HlU1muxd0r83R7NHnGRSw
        include base-4.14.1.0
        include boxes-0.1.5-6yGh7Ph1Utu8wxVXCPSyeh
        include bytestring-0.10.12.0
        include colour-2.3.5-C29WD8Qg69AL3qeQgayHy7
        include containers-0.6.2.1
        include data-default-0.7.1.1-CYtyoeRTyO02nIfJy16t2P
        include diagrams-lib-1.4.3-4T3pMpxqmsxGUSWBTZP2Yh
        include diagrams-svg-1.4.3-FUc4JukGvmFI4bl5xnYPyB
        include directory-1.3.6.0
        include exceptions-0.10.4
        include filepath-1.4.2.1
        include fmt-0.6.1.2-D5ieS3bbHQnG2n449eoYuO
        include formatting-6.3.7-8emLtPUrsOY3pXsJkNgaCF
        include hashable-1.3.0.0-KzryYcDwvBoH5VSVaB6Zwd
        include lens-4.19.2-7Erpllp4x6QJde6QNgn5al
        include mtl-2.2.2
        include optparse-simple-0.1.1.3-GCsoEruwqgiElMemPrtAbu
        include orgmode-parse-0.2.3-GD8YuLIKYQFIZjgiqbOh4W
        include process-1.6.9.0
        include text-1.2.4.1
        include time-1.9.3
        include turtle-1.5.21-IR6Ez24zwnKHcviOCCfKnL
        include universum-1.7.2-4C67qAn8hpSBO1f4hPkb4C
        include yaml-0.11.5.0-GXKwO3znmdcIxQZ2p4lFUb
        OrgStat.Ast=orgstat-0.1.9-Irl6FpkNJ5D124nqwx7n2s:OrgStat.Ast,OrgStat.CLI=orgstat-0.1.9-Irl6FpkNJ5D124nqwx7n2s:OrgStat.CLI,OrgStat.Config=orgstat-0.1.9-Irl6FpkNJ5D124nqwx7n2s:OrgStat.Config,OrgStat.Helpers=orgstat-0.1.9-Irl6FpkNJ5D124nqwx7n2s:OrgStat.Helpers,OrgStat.IO=orgstat-0.1.9-Irl6FpkNJ5D124nqwx7n2s:OrgStat.IO,OrgStat.Logging=orgstat-0.1.9-Irl6FpkNJ5D124nqwx7n2s:OrgStat.Logging,OrgStat.Logic=orgstat-0.1.9-Irl6FpkNJ5D124nqwx7n2s:OrgStat.Logic,OrgStat.Outputs=orgstat-0.1.9-Irl6FpkNJ5D124nqwx7n2s:OrgStat.Outputs,OrgStat.Outputs.Block=orgstat-0.1.9-Irl6FpkNJ5D124nqwx7n2s:OrgStat.Outputs.Block,OrgStat.Outputs.Class=orgstat-0.1.9-Irl6FpkNJ5D124nqwx7n2s:OrgStat.Outputs.Class,OrgStat.Outputs.Script=orgstat-0.1.9-Irl6FpkNJ5D124nqwx7n2s:OrgStat.Outputs.Script,OrgStat.Outputs.Summary=orgstat-0.1.9-Irl6FpkNJ5D124nqwx7n2s:OrgStat.Outputs.Summary,OrgStat.Outputs.Timeline=orgstat-0.1.9-Irl6FpkNJ5D124nqwx7n2s:OrgStat.Outputs.Timeline,OrgStat.Outputs.Types=orgstat-0.1.9-Irl6FpkNJ5D124nqwx7n2s:OrgStat.Outputs.Types,OrgStat.Parser=orgstat-0.1.9-Irl6FpkNJ5D124nqwx7n2s:OrgStat.Parser,OrgStat.Scope=orgstat-0.1.9-Irl6FpkNJ5D124nqwx7n2s:OrgStat.Scope,OrgStat.Util=orgstat-0.1.9-Irl6FpkNJ5D124nqwx7n2s:OrgStat.Util,OrgStat.WorkMonad=orgstat-0.1.9-Irl6FpkNJ5D124nqwx7n2s:OrgStat.WorkMonad,Paths_orgstat=orgstat-0.1.9-Irl6FpkNJ5D124nqwx7n2s:Paths_orgstat
    unit orgstat-0.1.9-9vAWSyx0oj1GxDiGvBvCm0-orgstat
        include base-4.14.1.0
        include bytestring-0.10.12.0
        include directory-1.3.6.0
        include exceptions-0.10.4
        include filepath-1.4.2.1
        include formatting-6.3.7-8emLtPUrsOY3pXsJkNgaCF
        include optparse-simple-0.1.1.3-GCsoEruwqgiElMemPrtAbu
        include orgstat-0.1.9-Irl6FpkNJ5D124nqwx7n2s
        include universum-1.7.2-4C67qAn8hpSBO1f4hPkb4C
    unit orgstat-0.1.9-BZzeApBR7Mk86xSdbWZLOo-orgstatarch
        include attoparsec-0.13.2.5-3HlU1muxd0r83R7NHnGRSw
        include base-4.14.1.0
        include bytestring-0.10.12.0
        include directory-1.3.6.0
        include exceptions-0.10.4
        include filepath-1.4.2.1
        include formatting-6.3.7-8emLtPUrsOY3pXsJkNgaCF
        include optparse-simple-0.1.1.3-GCsoEruwqgiElMemPrtAbu
        include orgmode-parse-0.2.3-GD8YuLIKYQFIZjgiqbOh4W
        include orgstat-0.1.9-Irl6FpkNJ5D124nqwx7n2s
        include text-1.2.4.1
        include time-1.9.3
        include turtle-1.5.21-IR6Ez24zwnKHcviOCCfKnL
        include universum-1.7.2-4C67qAn8hpSBO1f4hPkb4C
        include unordered-containers-0.2.13.0-LfX1QsW45a49ihvoGPwDYs
    unit orgstat-0.1.9-DitnakZGhXaFkCDXafAtcx-orgstat-test
        include HUnit-1.6.2.0-H4jVFZwCpZD3gLDGn5X1iO
        include QuickCheck-2.14.2-emyW1cBTLx2Wxf2sOiMGt
        include base-4.14.1.0
        include colour-2.3.5-C29WD8Qg69AL3qeQgayHy7
        include hspec-2.7.8-Aj0ANgyNRIsCgvQsvGCAl8
        include lens-4.19.2-7Erpllp4x6QJde6QNgn5al
        include orgstat-0.1.9-Irl6FpkNJ5D124nqwx7n2s
        include quickcheck-text-0.1.2.1-9ONdHqB9s1mEZeV5Jvlbp8
        include text-1.2.4.1
        include time-1.9.3
        include transformers-0.5.6.2
        include universum-1.7.2-4C67qAn8hpSBO1f4hPkb4C
Ready component graph:
    definite orgstat-0.1.9-Irl6FpkNJ5D124nqwx7n2s
        depends aeson-1.5.6.0-EJQ3uwDcV54EcRrluu2VE
        depends ansi-terminal-0.11-9FsdYO0Hps83uFNLVimBZk
        depends attoparsec-0.13.2.5-3HlU1muxd0r83R7NHnGRSw
        depends base-4.14.1.0
        depends boxes-0.1.5-6yGh7Ph1Utu8wxVXCPSyeh
        depends bytestring-0.10.12.0
        depends colour-2.3.5-C29WD8Qg69AL3qeQgayHy7
        depends containers-0.6.2.1
        depends data-default-0.7.1.1-CYtyoeRTyO02nIfJy16t2P
        depends diagrams-lib-1.4.3-4T3pMpxqmsxGUSWBTZP2Yh
        depends diagrams-svg-1.4.3-FUc4JukGvmFI4bl5xnYPyB
        depends directory-1.3.6.0
        depends exceptions-0.10.4
        depends filepath-1.4.2.1
        depends fmt-0.6.1.2-D5ieS3bbHQnG2n449eoYuO
        depends formatting-6.3.7-8emLtPUrsOY3pXsJkNgaCF
        depends hashable-1.3.0.0-KzryYcDwvBoH5VSVaB6Zwd
        depends lens-4.19.2-7Erpllp4x6QJde6QNgn5al
        depends mtl-2.2.2
        depends optparse-simple-0.1.1.3-GCsoEruwqgiElMemPrtAbu
        depends orgmode-parse-0.2.3-GD8YuLIKYQFIZjgiqbOh4W
        depends process-1.6.9.0
        depends text-1.2.4.1
        depends time-1.9.3
        depends turtle-1.5.21-IR6Ez24zwnKHcviOCCfKnL
        depends universum-1.7.2-4C67qAn8hpSBO1f4hPkb4C
        depends yaml-0.11.5.0-GXKwO3znmdcIxQZ2p4lFUb
    definite orgstat-0.1.9-DitnakZGhXaFkCDXafAtcx-orgstat-test
        depends HUnit-1.6.2.0-H4jVFZwCpZD3gLDGn5X1iO
        depends QuickCheck-2.14.2-emyW1cBTLx2Wxf2sOiMGt
        depends base-4.14.1.0
        depends colour-2.3.5-C29WD8Qg69AL3qeQgayHy7
        depends hspec-2.7.8-Aj0ANgyNRIsCgvQsvGCAl8
        depends lens-4.19.2-7Erpllp4x6QJde6QNgn5al
        depends orgstat-0.1.9-Irl6FpkNJ5D124nqwx7n2s
        depends quickcheck-text-0.1.2.1-9ONdHqB9s1mEZeV5Jvlbp8
        depends text-1.2.4.1
        depends time-1.9.3
        depends transformers-0.5.6.2
        depends universum-1.7.2-4C67qAn8hpSBO1f4hPkb4C
    definite orgstat-0.1.9-BZzeApBR7Mk86xSdbWZLOo-orgstatarch
        depends attoparsec-0.13.2.5-3HlU1muxd0r83R7NHnGRSw
        depends base-4.14.1.0
        depends bytestring-0.10.12.0
        depends directory-1.3.6.0
        depends exceptions-0.10.4
        depends filepath-1.4.2.1
        depends formatting-6.3.7-8emLtPUrsOY3pXsJkNgaCF
        depends optparse-simple-0.1.1.3-GCsoEruwqgiElMemPrtAbu
        depends orgmode-parse-0.2.3-GD8YuLIKYQFIZjgiqbOh4W
        depends orgstat-0.1.9-Irl6FpkNJ5D124nqwx7n2s
        depends text-1.2.4.1
        depends time-1.9.3
        depends turtle-1.5.21-IR6Ez24zwnKHcviOCCfKnL
        depends universum-1.7.2-4C67qAn8hpSBO1f4hPkb4C
        depends unordered-containers-0.2.13.0-LfX1QsW45a49ihvoGPwDYs
    definite orgstat-0.1.9-9vAWSyx0oj1GxDiGvBvCm0-orgstat
        depends base-4.14.1.0
        depends bytestring-0.10.12.0
        depends directory-1.3.6.0
        depends exceptions-0.10.4
        depends filepath-1.4.2.1
        depends formatting-6.3.7-8emLtPUrsOY3pXsJkNgaCF
        depends optparse-simple-0.1.1.3-GCsoEruwqgiElMemPrtAbu
        depends orgstat-0.1.9-Irl6FpkNJ5D124nqwx7n2s
        depends universum-1.7.2-4C67qAn8hpSBO1f4hPkb4C
Using Cabal-3.2.1.0 compiled by ghc-8.10
Using compiler: ghc-8.10.4
Using install prefix:
/nix/store/v61wf8h5cgan84y3qryvrs2qiqnq5vim-orgstat-0.1.9
Executables installed in:
/nix/store/v61wf8h5cgan84y3qryvrs2qiqnq5vim-orgstat-0.1.9/bin
Libraries installed in:
/nix/store/v61wf8h5cgan84y3qryvrs2qiqnq5vim-orgstat-0.1.9/lib/ghc-8.10.4/x86_64-linux-ghc-8.10.4/orgstat-0.1.9-Irl6FpkNJ5D124nqwx7n2s
Dynamic Libraries installed in:
/nix/store/v61wf8h5cgan84y3qryvrs2qiqnq5vim-orgstat-0.1.9/lib/ghc-8.10.4/x86_64-linux-ghc-8.10.4
Private executables installed in:
/nix/store/v61wf8h5cgan84y3qryvrs2qiqnq5vim-orgstat-0.1.9/libexec/x86_64-linux-ghc-8.10.4/orgstat-0.1.9
Data files installed in:
/nix/store/v61wf8h5cgan84y3qryvrs2qiqnq5vim-orgstat-0.1.9/share/x86_64-linux-ghc-8.10.4/orgstat-0.1.9
Documentation installed in:
/nix/store/c734gdxwq1s83w1hlq6dwzq2rppcmp14-orgstat-0.1.9-doc/share/doc/orgstat-0.1.9
Configuration files installed in:
/nix/store/v61wf8h5cgan84y3qryvrs2qiqnq5vim-orgstat-0.1.9/etc
No alex found
Using ar found on system at:
/nix/store/xdii8qvch5h8chyp0z2is2qzky565w68-binutils-2.35.1/bin/ar
No c2hs found
No cpphs found
No doctest found
Using gcc version 10.2.0 given by user at:
/nix/store/pqiwg1jw0s7qzdi4m9xbps2jb98fsxbx-gcc-wrapper-10.2.0/bin/gcc
Using ghc version 8.10.4 found on system at:
/nix/store/qabwi7rkqwppxbp575vq7khy6v2vqv81-ghc-8.10.4/bin/ghc
Using ghc-pkg version 8.10.4 found on system at:
/nix/store/qabwi7rkqwppxbp575vq7khy6v2vqv81-ghc-8.10.4/bin/ghc-pkg
No ghcjs found
No ghcjs-pkg found
No greencard found
Using haddock version 2.24.0 found on system at:
/nix/store/qabwi7rkqwppxbp575vq7khy6v2vqv81-ghc-8.10.4/bin/haddock
No happy found
Using haskell-suite found on system at: haskell-suite-dummy-location
Using haskell-suite-pkg found on system at: haskell-suite-pkg-dummy-location
No hmake found
Using hpc version 0.68 found on system at:
/nix/store/qabwi7rkqwppxbp575vq7khy6v2vqv81-ghc-8.10.4/bin/hpc
Using hsc2hs version 0.68.7 found on system at:
/nix/store/qabwi7rkqwppxbp575vq7khy6v2vqv81-ghc-8.10.4/bin/hsc2hs
Using hscolour version 1.24 found on system at:
/nix/store/f1pl8nx2k3pfpwnsxvbrggsfmmcm1d4p-hscolour-1.24.4/bin/HsColour
No jhc found
Using ld found on system at:
/nix/store/r7v3x5a68a4mxc2kh04cmyzx7xq00w2g-binutils-wrapper-2.35.1/bin/ld.gold
No pkg-config found
Using runghc version 8.10.4 found on system at:
/nix/store/qabwi7rkqwppxbp575vq7khy6v2vqv81-ghc-8.10.4/bin/runghc
Using strip version 2.35 found on system at:
/nix/store/xdii8qvch5h8chyp0z2is2qzky565w68-binutils-2.35.1/bin/strip
Using tar found on system at:
/nix/store/hxpycgfpdnmnc9iwxakxr037k0bvdq97-gnutar-1.32/bin/tar
No uhc found
building
Preprocessing library for orgstat-0.1.9..
Building library for orgstat-0.1.9..
[ 1 of 19] Compiling OrgStat.Ast      ( src/OrgStat/Ast.hs, dist/build/OrgStat/Ast.o, dist/build/OrgStat/Ast.dyn_o )
[ 2 of 19] Compiling OrgStat.CLI      ( src/OrgStat/CLI.hs, dist/build/OrgStat/CLI.o, dist/build/OrgStat/CLI.dyn_o )
[ 3 of 19] Compiling OrgStat.Logging  ( src/OrgStat/Logging.hs, dist/build/OrgStat/Logging.o, dist/build/OrgStat/Logging.dyn_o )
[ 4 of 19] Compiling OrgStat.Outputs.Types ( src/OrgStat/Outputs/Types.hs, dist/build/OrgStat/Outputs/Types.o, dist/build/OrgStat/Outputs/Types.dyn_o )
[ 5 of 19] Compiling OrgStat.Outputs.Class ( src/OrgStat/Outputs/Class.hs, dist/build/OrgStat/Outputs/Class.o, dist/build/OrgStat/Outputs/Class.dyn_o )
[ 6 of 19] Compiling OrgStat.Parser   ( src/OrgStat/Parser.hs, dist/build/OrgStat/Parser.o, dist/build/OrgStat/Parser.dyn_o )
[ 7 of 19] Compiling OrgStat.Scope    ( src/OrgStat/Scope.hs, dist/build/OrgStat/Scope.o, dist/build/OrgStat/Scope.dyn_o )
[ 8 of 19] Compiling OrgStat.Util     ( src/OrgStat/Util.hs, dist/build/OrgStat/Util.o, dist/build/OrgStat/Util.dyn_o )
[ 9 of 19] Compiling OrgStat.Outputs.Timeline ( src/OrgStat/Outputs/Timeline.hs, dist/build/OrgStat/Outputs/Timeline.o, dist/build/OrgStat/Outputs/Timeline.dyn_o )

src/OrgStat/Outputs/Timeline.hs:226:22: error:
    • Couldn't match type ‘[]’ with ‘NonEmpty’
      Expected type: NonEmpty LocalTime
        Actual type: [LocalTime]
    • In the first argument of ‘minimum’, namely ‘(map cFrom c)’
      In the expression: minimum (map cFrom c)
      In the expression: (minimum (map cFrom c), maximum (map cTo c))
    |
226 |         in (minimum (map cFrom c), maximum (map cTo c))
    |                      ^^^^^^^^^^^

src/OrgStat/Outputs/Timeline.hs:226:45: error:
    • Couldn't match type ‘[]’ with ‘NonEmpty’
      Expected type: NonEmpty LocalTime
        Actual type: [LocalTime]
    • In the first argument of ‘maximum’, namely ‘(map cTo c)’
      In the expression: maximum (map cTo c)
      In the expression: (minimum (map cFrom c), maximum (map cTo c))
    |
226 |         in (minimum (map cFrom c), maximum (map cTo c))
    |                                             ^^^^^^^^^
[10 of 19] Compiling OrgStat.Outputs.Block ( src/OrgStat/Outputs/Block.hs, dist/build/OrgStat/Outputs/Block.o, dist/build/OrgStat/Outputs/Block.dyn_o )
[11 of 19] Compiling OrgStat.Config   ( src/OrgStat/Config.hs, dist/build/OrgStat/Config.o, dist/build/OrgStat/Config.dyn_o )
[12 of 19] Compiling OrgStat.IO       ( src/OrgStat/IO.hs, dist/build/OrgStat/IO.o, dist/build/OrgStat/IO.dyn_o )
[13 of 19] Compiling OrgStat.WorkMonad ( src/OrgStat/WorkMonad.hs, dist/build/OrgStat/WorkMonad.o, dist/build/OrgStat/WorkMonad.dyn_o )
[14 of 19] Compiling OrgStat.Helpers  ( src/OrgStat/Helpers.hs, dist/build/OrgStat/Helpers.o, dist/build/OrgStat/Helpers.dyn_o )
[15 of 19] Compiling OrgStat.Outputs.Summary ( src/OrgStat/Outputs/Summary.hs, dist/build/OrgStat/Outputs/Summary.o, dist/build/OrgStat/Outputs/Summary.dyn_o )
[16 of 19] Compiling OrgStat.Outputs.Script ( src/OrgStat/Outputs/Script.hs, dist/build/OrgStat/Outputs/Script.o, dist/build/OrgStat/Outputs/Script.dyn_o )
[19 of 19] Compiling Paths_orgstat    ( dist/build/autogen/Paths_orgstat.hs, dist/build/Paths_orgstat.o, dist/build/Paths_orgstat.dyn_o )
builder for '/nix/store/gk8fb9ksd0z464zbcvb164aabrpq94nc-orgstat-0.1.9.drv' failed with exit code 1
error: build of '/nix/store/gk8fb9ksd0z464zbcvb164aabrpq94nc-orgstat-0.1.9.drv' failed
```

**Comment by volhovm** [comment]
4.14 is not supported: `base >=4.13 && <4.14` is in `orgstat.cabal`. Not sure why nixpkgs attempts to build with it anyway.

**Comment by sternenseemann** [comment]
stackage-nightly has base 4.14 and we follow that. Due to propagation it is not always possible to have multiple versions of the same package in the package set unfortunately.

We know about the breakage and `orgstat` is marked as broken due to that currently, I was just looking for a way to make it work again.

**Comment by volhovm** [comment]
@sternenseemann I've encountered a problem: `diagrams-lib` and `diagrams-svg` work with `base-4.13` at most, and they have not been updated for a few years. They are thus unavailable with the new LTS (17) -- I'm not sure I understand how one would include a sub-package built with another set of versions of dependencies the main project has. Alternatively, I have tried to somehow add `base-4.14` without changing LTS, but without success. Do you have an idea of what the solution could be here? (switching to a different plotting library is the worst case, so I'm looking for something simpler)

**Comment by sternenseemann** [comment]
`diagrams-lib` is not really an issue since they are compatible with base 4.13, but they just haven't released a version with updated bounds yet unfortunately. I've fixed diagrams-lib for the next haskell update in nixpkgs, so it should work on the `haskell-updates` branch now and on `master` after friday: <https://github.com/NixOS/nixpkgs/pull/117419>. `diagrams-svg` also builds there currently.

We probably should annoy upstream to release updated bounds to hackage.

**Comment by volhovm** [comment]
I think it's easy to fix, but what's the easiest way to test it? I only use stack, so do you have any idea how to compile all dependencies with 4.13, but the project itself with 4.14?..

**Comment by sternenseemann** [comment]
I'd say update to a nightly resolver like `nightly-2020-12-14` which has base 4.14. `nixpkgs` switched to nightly over lts a few months ago that is probably how we arrived in this situation.

**Comment by sternenseemann** [comment]
Nevermind, `stackage-nightly` trips up on diagrams. Try this:

```nix
let
   pkgs = import (builtins.fetchTarball {
      url = "https://github.com/nixos/nixpkgs/archive/fe0bdf4d0178bf7420c417a3aae44b369feec6e5.tar.gz";
      sha256 = "1yh3lig5srfc3ln3qsic2i091p7rdci3rgmhv1dyks2q67ihi4c7";
   }) { };

   inherit (pkgs)
      haskell
      haskellPackages
      ;
in
   haskell.lib.markUnbroken (haskell.lib.overrideSrc haskellPackages.orgstat {
      version = "git";
      src = ./.;
   })
```

(and then just nix-build that or nix-shell -A env)

**Comment by volhovm** [comment]
Sorry for the delay. I fixed compatibility with `base-4.14` now, the new orgstat version is `0.1.10`. Thank you for providing the build script! Tell me if this works, I'll close the issue for now (feel free to reopen).

**Comment by sternenseemann** [comment]
Thanks, that fixes it indeed! Should be working in nixpkgs again friday to saturday :)
